src/vm/jitinterfacegen.cpp

```diff
@@ -224,10 +224,10 @@ void InitJITHelpers1()
         if (GCHeap::UseAllocationContexts())
         {
 #ifdef FEATURE_IMPLICIT_TLS
-            SetJitHelperFunction(CORINFO_HELP_NEWSFAST, JIT_NewS_MP_FastPortable);
-            SetJitHelperFunction(CORINFO_HELP_NEWSFAST_ALIGN8, JIT_NewS_MP_FastPortable);
-            SetJitHelperFunction(CORINFO_HELP_NEWARR_1_VC, JIT_NewArr1VC_MP_FastPortable);
-            SetJitHelperFunction(CORINFO_HELP_NEWARR_1_OBJ, JIT_NewArr1OBJ_MP_FastPortable);
+            SetJitHelperFunction(CORINFO_HELP_NEWSFAST, reinterpret_cast<void*>(JIT_NewS_MP_FastPortable));
```

Comment:
I read it from Jackson books.

If a function is called from external entities (i.e. something called a library) we check parameters in entry to a function. For internally controlled and called functions/macros, we assert correctness of passed arguments before calling them.

This is my reasoning.